### PR TITLE
chore: add database-related settings

### PR DESCRIPTION
**Pull request by alan910127**
---

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request adds a new `db.py` module, modifies the `app.py` and `config.py` modules, and introduces a new `compose.yml` file. It also updates the `pdm.lock` and `pyproject.toml` files to include the necessary dependencies.
> 
> ## What changed
> - Added a new `db.py` module that includes functions for initializing and closing a database connection pool, as well as a function for retrieving a database connection.
> - Modified the `app.py` module to import the `db` module and include a new `lifespan` async context manager that initializes and closes the database connection pool.
> - Modified the `config.py` module to include a new `DATABASE_URL` setting for specifying the PostgreSQL database URL.
> - Added a new `compose.yml` file that defines a Docker Compose service for a PostgreSQL database.
> - Updated the `pdm.lock` and `pyproject.toml` files to include the `asyncpg` dependency.
> 
> ## How to test
> 1. Ensure that Docker is installed on your system.
> 2. Run the `docker-compose up -d` command to start the PostgreSQL database service.
> 3. Run the application and verify that the database connection is successfully initialized and closed.
> 4. Verify that the `DATABASE_URL` setting in the `config.py` module is correctly set to the PostgreSQL database URL.
> 
> ## Why make this change
> - The addition of the `db.py` module allows for better separation of concerns and encapsulation of database-related functionality.
> - The modification of the `app.py` module introduces an async context manager for managing the lifespan of the application, including the initialization and closing of the database connection pool.
> - The addition of the `DATABASE_URL` setting in the `config.py` module allows for easy configuration of the PostgreSQL database URL.
> - The inclusion of the `compose.yml` file simplifies the setup of a PostgreSQL database service for local development and testing.
> - The update to the `pdm.lock` and `pyproject.toml` files ensures that the `asyncpg` dependency is included and installed correctly.
</details>

### CODE REVIEW

**Comment by alan910127** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/3?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #4** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/4?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/5?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
          * **PR #6** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/6?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/5?utm_source=stack-comment).

File: app/app.py [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 from app.core import settings
 
 logger.setup_logging(
     development=settings.ENVIRONMENT == "development",
     log_level=settings.LOG_LEVEL,
 )
 
+
+@asynccontextmanager
+async def lifespan(_app: FastAPI):
```
Comment (alan910127) [inline app/app.py:18]: to connect on start/disconnect on close

File: app/db.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+async def close_pool():
+    global pool
+    if pool is not None:
+        await pool.close()
+        pool = None
+
+
+async def get_db() -> AsyncGenerator[Connection, None]:
```
Comment (alan910127) [inline app/db.py:31]: this will be used as a FastAPI dependency function

**Review by BaBa0525 — APPROVED** [review]
[birdbox](https://app.graphite.dev/macros/GymSquad)

**Comment by alan910127** [comment]
### Merge activity

* **Jan 27, 6:16 AM**: @alan910127 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/5).
* **Jan 27, 6:21 AM**: [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/5) rebased this pull request as part of a merge.
* **Jan 27, 6:22 AM**: @alan910127 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/5).

**Pull request merged**